tests/integration/client/sdk/v1/test_records.py

```diff
@@ -54,6 +55,50 @@ def test_dataset():
 
 @pytest.mark.asyncio
 class TestRecordsSDK:
+    @pytest.mark.parametrize("role", [UserRole.owner, UserRole.admin])
+    async def test_update_record_with_vectors(self, owner: User, role: UserRole) -> None:
+        user = await UserFactory.create(role=role)
+
+        api.init(api_key=owner.api_key)
```

Comment:
We should avoid using test factories for integration tests. 

```suggestion
    def test_update_record_with_vectors(self, owner: User, role: UserRole) -> None:
        api.init(api_key=owner.api_key)
        user = User.create(username="user", role=role,...)
```